src/main.cpp

```diff
@@ -19,147 +19,168 @@
-    po::positional_options_description p;
-    p.add("gbdxm-file", -1);
+        string action = argv[1];
+        to_lower(action);
 
-    // Parse the arguments
-    po::variables_map vm;
+        // Add gbdxm-file as a positional argument for convenience
+        po::positional_options_description p;
+        p.add("gbdxm-file", -1);
 
-    try {
-        po::store(po::command_line_parser(argc - 1, &argv[1])
-            .extra_style_parser(&po::ignore_numbers)
-            .options(all)
-            .positional(p)
-            .run(), vm);
-        po::notify(vm);
-    } catch(const exception& e) {
-        cerr << e.what() << endl;
-        exit(1);
-    } catch(...) {
-        cerr << "Unknown error parsing command line arguments." << endl;
-        exit(1);
-    }
+        // Parse the arguments
+        po::variables_map vm;
 
-    // Read arguments, if invalid action or "help", print the usage details
-    unique_ptr<GbdxmArgs> args(readArgs(vm, action));
-    if(!args) {
-        if(action == "help") {
+        try {
+            po::store(po::command_line_parser(argc - 1, &argv[1])
+                          .extra_style_parser(buildExtraStyleParser())
+                          .options(all)
+                          .positional(p)
+                          .run(), vm);
+            po::notify(vm);
+        } catch(...) {
             cout << visible << endl;
-            exit(0);
-        } else {
+            DG_ERROR_RETHROW("");
+        }
+
+        setupLogging(vm);
+
+        // Read arguments, if invalid action or "help", print the usage details
+        auto args = readArgs(vm, action);
+        if(!args) {
             cout << visible << endl;
-            cerr << "Invalid action. The correct actions are help, show, pack, and unpack." << endl << endl;
-            exit(1);
+            DG_CHECK(action == "help", "Invalid action. The correct actions are help, show, pack, and unpack");
+
+            exit(0);
         }
+
+        doAction(*args);
+    } catch (...) {
+        DG_ERROR_LOG(gbdxm, DG_ERROR_FROM_CURRENT(""));
+        return 1;
     }
 
-    doAction(*args);
+    return 0;
 }
 
-namespace dg { namespace deepcore { namespace classification {
+namespace dg { namespace gbdxm {
+
+po::detail::cmdline::style_parser buildExtraStyleParser()
+{
+    vector<po::detail::cmdline::style_parser> parsers;
+
+    parsers.emplace_back(&po::ignore_numbers);
+
+    for(const auto& type : classification::ModelIdentifier::types()) {
+        parsers.emplace_back(po::prefix_argument(type));
+    }
+
+    return po::combine_style_parsers(parsers);
+}
 
 po::options_description buildVisibleOptions()
 {
     po::options_description desc(
-        "GDBX Model Packaging Tool\n"
+        "GBDX Model Packaging Tool\n"
         "Version: " GBDXM_VERSION_STRING "\n\n"
         "Usage: gbdxm <action> [options] [gbdxm file]\n\n"
         "Actions:\n"
         "  help\t\t\t Show this help message.\n"
```

Comment:
Since tab stops are set by the tty, this should have spaces in the string, use a formatter, or use spaces for column alignment.  The last option would be:
```
"  help  \t\t Show this help message.\n"
"  show  \t\t Show package metadata.\n"
"  pack  \t\t Pack a model into a GDBX package.\n"
"  unpack\t\t Unpack a GBDX package and output the original model.\n\n"
```